File: rb/features.rb
# SpiderManMovies SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module SpiderManMoviesFeatures
  def self.make_feature(name)
    case name
    when "base"
      SpiderManMoviesBaseFeature.new
    when "ratelimit"
      SpiderManMoviesRatelimitFeature.new
    when "retry"
      SpiderManMoviesRetryFeature.new
    when "test"
      SpiderManMoviesTestFeature.new
    when "timeout"
      SpiderManMoviesTimeoutFeature.new
    else
      SpiderManMoviesBaseFeature.new
    end
  end
end

File: py/features.py
# SpiderManMovies SDK feature factory

from feature.base_feature import SpiderManMoviesBaseFeature
from feature.test_feature import SpiderManMoviesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SpiderManMoviesBaseFeature(),
        "test": lambda: SpiderManMoviesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
